File: app/applet/test_api.py
import sys
import urllib.request
import urllib.error

urls = [
    'http://localhost:3000/api/products',
    'http://localhost:3000/api/categories',
    'http://localhost:3000/api/settings',
    'http://localhost:3000/api/hero-banners',
    'http://localhost:3000/api/brands',
    'http://localhost:3000/api/admin/products',
    'http://localhost:3000/api/admin/settings'
]

for url in urls:
    try:
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        with urllib.request.urlopen(req) as response:
            print(f"[{url}] {response.status}")
    except urllib.error.HTTPError as e:
        body = e.read().decode('utf-8')
        print(f"[{url}] {e.code}: {body[:500]}")
    except urllib.error.URLError as e:
        print(f"[{url}] URL Error: {e.reason}")

File: test_api.js
import http from 'http';

function testEndpoint(path) {
  http.get('http://localhost:3000' + path, (res) => {
    let data = '';
    res.on('data', (chunk) => data += chunk);
    res.on('end', () => {
      console.log(`[${path}] Status: ${res.statusCode}`);
      if (res.statusCode >= 400) {
        console.log(`[${path}] Error Response: ${data.substring(0, 500)}`);
      }
    });
  }).on('error', (err) => {
    console.log(`[${path}] Request Error: ${err.message}`);
  });
}

testEndpoint('/api/products');
testEndpoint('/api/categories');
testEndpoint('/api/settings');
testEndpoint('/api/hero-banners');
testEndpoint('/api/brands');
testEndpoint('/api/admin/products');
testEndpoint('/api/admin/settings');
